template/requirements.txt

```diff
@@ -1,17 +1,11 @@
 zenml
 torch>=2.2.0
-lightning @ git+https://github.com/Lightning-AI/lightning@ed367ca675861cdf40dbad2e4d66f7eee2ec50af
-jsonargparse[signatures]  # CLI
-bitsandbytes==0.41.0      # quantization
-scipy                     # required by bitsandbytes
-sentencepiece             # llama-based models
-tokenizers                # pythia, falcon, redpajama
-datasets                  # eval
-requests                  # scripts/prepare_*
-zstandard                 # scripts/prepare_redpajama.py, scripts/prepare_starcoder.py
-pandas                    # scripts/prepare_csv.py, scripts/prepare_starcoder.py
-pyarrow                   # scripts/prepare_starcoder.py
-# eval
-git+https://github.com/EleutherAI/lm-evaluation-harness.git@115206dc89dad67b8beaa90051fb52db77f0a529
-# scripts/prepare_slimpajama.py, scripts/prepare_starcoder.py, pretrain/tinyllama.py
-lightning[data] @ git+https://github.com/Lightning-AI/lightning@ed367ca675861cdf40dbad2e4d66f7eee2ec50af
+datasets<=2.18
+transformers
+peft
+bitsandbytes==0.41.3
+scipy
+evaluate
+rouge_score
+nltk
+accelerate==0.29.2
```

Comment:
do later versions break things? I'm somewhat worried that in 6 months this example will be really out of date (since `accelerate` library is constantly releasing)

Reply:
Maybe not, or not yet 🙂 
Datasets break things if upgraded, I will try relaxing it and datasets

Reply:
Relaxed